src/main/java/Deadline.java

```diff
@@ -0,0 +1,20 @@
+public class Deadline extends Task {
+    /** To check if the task is a deadline task */
+    protected boolean isDeadline;
+
+    public Deadline(String description) {
+        super(description);
+        this.isDeadline = false;
+    }
+
+    @Override
+    public String getStatusOfTypeTask() {
+        //mark Task with Deadline with "D"
+        return (isDeadline ? "D" : " ");
```

Comment:
Avoid complicated expressions, can consider breaking down the logic into multiple statements.